### DISCUSSION TITLE
My comments of yesterday!!!!

### DISCUSSION THREAD

**Discussion by midimusic** [GrandOrgue/grandorgue#681] [General]
The comments I made yesterday in "General" seem to have disappeared but thanks Lars and Oleg for your responses I received by email. The discussion area has filled up with dozens of unanswered messages all at the same time so it is next to impossible to find anything.

However your answers did not help much!
Lars: your list of dependencies would be useful but wx-glib3 and wavepack appear to be unavailable for ubuntu so how did my previous successful installation find them? So how do I get them? The others appear to be already installed.

Oleg: The downloads I made came from the opensuse website and there is no mention of dependencies. Your suggestion to read the github readme indicated that a package of dependencies was available but did not indicate where to find it.

In fact it seems to be all rather a mess at the moment so I will give up for now until you have sorted it out.

csw900

**Comment by rousseldenis** [comment]
Hi @midimusic.

I understand your frustration. But, we are working for the best. So, some actions can make things a little bit confused (temporarily).

For your install problem, you can for sure follow Oleg readme suggestions.

To find what to install, you can go to the main project page :

![Sélection_657](https://user-images.githubusercontent.com/19529533/133663422-877b17ef-5591-47a5-a56b-3ec797883232.png)

There, you'll find the different files you need. Don't hesitate to come back if you have questions

**Reply by rousseldenis** [reply]
@midimusic You can find the thread you opened yesterday there: https://github.com/GrandOrgue/grandorgue/discussions/31

**Comment by oleg68** [comment]
@midimusic 

> Oleg: The downloads I made came from the opensuse website and there is no mention of dependencies. Your suggestion to read the github readme indicated that a package of dependencies was available but did not indicate where to find it.

This url is indicated in the [INSTALL.md](https://github.com/GrandOrgue/grandorgue/blob/master/INSTALL.md)

```
Download installation files from Github.
For a release version go to https://github.com/GrandOrgue/grandorgue/releases and select a release number.
For an intermediate build go to https://github.com/GrandOrgue/grandorgue/actions, select a build and scroll down to Artifacts The necessary file names to download depend on the target operation system.
```

**Comment by larspalo** [comment]
I have wavpack, libwavpack1 and libwavpack1-dev installed (-dev is for building, you might not need it)

I also have libwxbase3.0-0v5, -dev, libwxgtk3.0-gtk3-0v5, -dev, wx-common, wx3.0-headers, wx3.0-i18n. You might not need all of them.

**Comment by midimusic** [comment]
Thanks all,

I found that the wx-gtk libs were already installed after I looked for the correct names. However libwavepack1 is not installed and is not available from the normal sources so I downloaded it from the wavepack website - it would not install because of unmet and unstated dependencies.
So I gave up and installed oleg's dependencies.deb package which went OK. However libwavepack1 is still not installed. I then tried again to install the GO file from opensuse. In spite of oleg's dependencies it would still not install because of unmet and unstated dependencies.
I then downloaded oleg's GO.deb and demo.deb files. These installed OK (they do have to be installed in the correct order, deps, GO and then demo) and GO is now working on my iMac.

I still do not know what GO's true dependencies are because the (graphic) installer never says. I did not install with apt, but this may give more useful information.

On testing GO with the full 3840X2160 resolution display with X2 magnification it does (in general) display correctly but the window frames do not - they are much too small. This is in common with many other apps on this system.

Good luck with getting it all sorted out,
csw900

**Reply by oleg68** [reply]
> I still do not know what GO's true dependencies are because the (graphic) installer never says. I did not install with apt, but this may give more useful information.

The only supported way of installing is using apt-get, that is described in the INSTALL.md. apt-get installs all necessary dependencies automatically, so the user does not need to know them.

**Reply by midimusic** [reply]
Oleg
You should realise that apt is not the only way to install .deb files. Ubuntu presents a graphical installer when you double click on a .deb file. This is the easiest option to use and normally works. The GO files in the opensuse (via a GO sourceforge link) present a graphic installer when you click on them. These files used to install correctly, but will no longer install because they now have unspecified and unmet dependencies.

apt will only install dependencies if they are available via the usual default repositories. I do not think that GO dependencies meet this requirement. If they do meet this need then why are you supplying a .deb file of dependencies which must be installed before GO will install?

Also why does your Github dependencies file not supply all the dependencies  that the opensuse GO installations need?

This is just a friendly reminder that there are a few things that need attending to. I know that you have put a lot of work into GO and things are now moving for the better after a long period of stagnation.
csw900

**Reply by oleg68** [reply]
> You should realise that apt is not the only way to install .deb files. Ubuntu presents a graphical installer when you double click on a .deb file. This is the easiest option to use and normally works. The GO files in the opensuse (via a GO sourceforge link) present a graphic installer when you click on them. 

The problem is that grandorgue.deb and grandorgue-resources.deb depend on each other so they must be installed simultaneusly. I'm not shure that it is possible with the double-click graphical installer. So I suggest to use apt-get.

> apt will only install dependencies if they are available via the usual default repositories. I do not think that GO dependencies meet this requirement. If they do meet this need then why are you supplying a .deb file of dependencies which must be installed before GO will install?

All depenndencies present in the default ubuntu repositories 18.04 or above. It has been tested.
